.github/workflows/tests.yml

```diff
@@ -571,6 +571,7 @@ jobs:
         env:
           POSTGRES: ${{ (matrix.database == 'Postgres') && 1 || '' }}
           WORKERS: ${{ (matrix.arrangement == 'workers') && 1 || '' }}
+          UNIX_SOCKETS: 1
         name: Run Complement Tests
 
```

Comment:
This PR looks good to me now :+1: 

I'm not sure how to solve or figure out the [typing problem](https://github.com/matrix-org/synapse/pull/15708#discussion_r1227437447). You can try raising in [#synapse-dev](https://matrix.to/#/#synapse-dev:matrix.org) to see if there is an interested party. Otherwise, I will ask internally.
 
And then we just need to remember to revert [`UNIX_SOCKETS: 1` change which already has a discussion tracking things](https://github.com/matrix-org/synapse/pull/15708#discussion_r1227426684)

Reply:
> And then we just need to remember to revert [`UNIX_SOCKETS: 1` change which already has a discussion tracking things](https://github.com/matrix-org/synapse/pull/15708#discussion_r1227426684)

I think this PR is ready to go (make this happen)

Reply:
Reverted in [0ae6036](https://github.com/matrix-org/synapse/pull/15708/commits/0ae603683fc135395fc882579b3da95074e84bcd)